sentry/Rakefile

```diff
@@ -62,26 +63,53 @@ task :update_projects do
   puts "There are #{sentry_projects.size} projects in Sentry"
   puts "There are #{apps.size} applications on GOV.UK"
 
-  apps.map do |app|
-    app_name = app.fetch("app_name")
-    app_slug = "app-#{app_name}"
+  apps.map {|app| create_or_update_projects(app, sentry_projects) }
+end
 
-    project = sentry_projects.find { |project| project.slug == app_slug }
+desc "Create or update Sentry project for a specific GOV.UK application"
+task :update_project, %i[project_name] => [] do |_, args|
+  project_name = args[:project_name]
 
-    if project
-      puts "√ #{app_name} has a Sentry project (#{project.slug})"
-    else
-      team_name = team_name(app["team"])
-      puts "Creating project #{app_name} (#{app_slug}) for team #{team_name}"
-      project = SentryApi.create_project(team_name, name: app_name, slug: app_slug)
-      puts "Created project for #{app_name} (#{project.slug})"
-    end
+  abort("Please provide a project name") if project_name.nil?
+
+  apps = fetch_govuk_apps
+  sentry_projects = SentryApi.organization_projects
+
+  apps_by_name = apps.select{ |app| app["app_name"] == project_name }
+  app = apps_by_name.first
+  abort("#{GOVUK_DOCS_APP_LIST} has more than one app called #{project_name}") if apps_by_name.count > 1
+  abort(app_not_found_warning(project_name)) unless apps.nil?
+  create_or_update_projects(app, sentry_projects)
```

Comment:
If we ensure `apps` is never `nil`, the only possibilities are that `apps_by_name` is:
- an empty array (no match with `project_name`)
- an array with one element (one match with `project_name`)
- an array with multiple elements (multiple matches with `project_name`)

We can then abort with meaningful messages if it's empty or with multiple elements, and continue if we only find the app we want.
```
  apps = fetch_govuk_apps || []
  sentry_projects = SentryApi.organization_projects
  apps_by_name = apps.select{ |app| app["app_name"] == project_name }

  abort("#{GOVUK_DOCS_APP_LIST} has more than one app called #{project_name}") if apps_by_name.count > 1
  abort(app_not_found_warning(project_name)) if apps_by_name.empty?

  create_or_update_projects(apps_by_name.first, sentry_projects)
```

Reply:
yeah good catch all, thanks!